cwlVersion: v1.2
class: CommandLineTool
doc: |
  This CWL tool runs Scalce for error correction of sequencing reads.
  For more information, please see https://sfu-compbio.github.io/scalce/

inputs:
  input:
    type: File
    inputBinding:
      position: 1
    doc: |
      Path to the input file containing sequencing reads. The file should be in
      either gzip-compressed or uncompressed format.
  kmer_length:
    type: int
    inputBinding:
      position: 2
    doc: |
      The length of k-mers to use for error correction. Must be a positive integer.
  output_prefix:
    type: string
    inputBinding:
      position: 3
      prefix: -o
    doc: |
      The prefix for output files generated by Scalce. The output files will be named
      ${output_prefix}_corrected.fastq.gz, ${output_prefix}_k${kmer_length}.txt, and
      ${output_prefix}_counts.txt.
  num_threads:
    type: int
    inputBinding:
      position: 4
      prefix: --threads
    default: 4
    doc: |
      The number of threads to use for processing. If not specified, defaults to 4.

outputs:
  output_fastq:
    type: File
    outputBinding:
      glob: "${output_prefix}_corrected.fastq.gz"

stdout: output.log
baseCommand: scalce
